models/instrument.py

```diff
@@ -1197,37 +1196,170 @@ def fiducial_aperture( cls ):
         # Note that this 5 is an index, not the value... it's coincidence that the index number is 5.
         return 5
 
+    # Gaia specific methods
+    # For GaiaDR3, catdata has fields:
+    # X_WORLD, Y_WORLD, MAG_G, MAGERR_G, MAG_BP, MAGERR_BP, MAG_RP, MAGERR_RP, STARPROB
 
     @classmethod
-    def get_GaiaDR3_transformation( cls, filter ):
+    def GaiaDR3_prune_star_cat(cls, catdata):
+        """Choose only rows from a catalog that have stars.
 
-        """Return a polynomial transformation from Gaia MAG_G to instrument magnitude.
+        Usually this is done by choosing a subset of the catalog
+        data with reasonable colors, and with a high enough STARPROB.
+        We also remove rows that have NaN values in any of the
+        critical columns (coordinates or magnitudes)
 
-        The returned array trns allows a conversion from Gaia MAG_G to
-        the magnitude through the desired filter using:
+        Parameters
+        ----------
+        catdata: dict or pandas.DataFrame or numpy.recarray or astropy.Table
+            Must be a data structure with the following keys:
+            MAG_G, MAGERR_G, MAG_BP, MAGERR_BP, MAG_RP, MAGERR_RP, STARPROB
+            The data structure, when indexed on those keys, should
+            return a 1D numpy array.
 
-          MAG_filter = Gaia_MAG_G - sum( trns[i] * ( Gaia_MAG _BP - Gaia_MAG_RP ) ** i )
+        Returns
+        -------
+        Returns a copy of the input data structure, with only the
+        rows that are likely to be stars.
+        """
+        output = copy.deepcopy(catdata)
+
+        # color cut
+        gaiaminbp_rp = 0.5
+        gaiamaxbp_rp = 3.0
+        color = catdata['MAG_BP'] - catdata['MAG_RP']
+        dex = (color >= gaiaminbp_rp) & (color <= gaiamaxbp_rp)
```

Comment:
...it's possible that we might potentially want to use different color cuts for different instruments.

For now I'd just document in the function docstring that it does this color cut, so that if somebody in the future has an instrument that needs a different color cut, they know they have to override this function in their instrument definition.

Reply:
I just put those parameters as function arguments with defaults. I don't know why I didn't just do that to begin with.